### PR TITLE
Set up a "Hello world" project as one app with Webpack

### PR DESCRIPTION
**Pull request by Uzair-Manzoor**
## This pull request has been created against implementing below mentioned requirements into the project:

- [x] Initialized the project with Git.
- [x] Installed necessary node dependencies.
- [x] Created a babel.config.js in the project's root folder.
- [x] Updated webpack.config.js to incorporate the babel loader.
- [x] Created a table for storing greeting messages.
- [x] Established an API endpoint for selecting a random greeting.
- [x] Created a static view to serve as the root of the application.
- [x] Developed the App component with react-router and integrated it into the static view.
- [x] Implemented the Greeting component to display greetings.
- [x] Established a store to connect to the API endpoint.
- [x] Added project documentation.

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  devtool: 'source-map',
+  entry: {
+    application: './app/javascript/application.js',
+  },
+  module: {
+    rules: [
+      {
+        test: /\.(js)$/,
+        exclude: /node_modules/,
+        use: ['babel-loader'],
```
Comment (souad988) [inline webpack.config.js:15]: Very well done by including the babel-loader

File: app/javascript/redux/greetings/greetingsSlice.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+export default (state = initialState, action) => {
+  switch (action.type) {
+    case `${FETCH}/fulfilled`:
+      return { ...state, greeting: action.payload };
+    default:
+      return state;
+  }
+};
+
+export const fetchGreeting = createAsyncThunk(FETCH, async () => {
```
Comment (souad988) [inline app/javascript/redux/greetings/greetingsSlice.js:20]: Well done 👍🏼 , the action to fetch greetings from backend is added

**Review by souad988 — APPROVED** [review]
Hi @Uzair-Manzoor , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Use the required gems
API endpoint for returning a random greeting was created
Create a database table for storing different greetings.

## Optional suggestions
Add steps to use docker to setup the app to your README since you created docker file.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/javascript/redux/greetings/greetingsSlice.js (hunk 2 above)
Comment (Uzair-Manzoor) [reply app/javascript/redux/greetings/greetingsSlice.js:20]: Thank you sincerely for the valuable feedback and kind appreciation; it has significantly elevated my motivation to new heights. 😊😊😊

**Pull request merged**